### PR TITLE
fix(uve): Update components in examples to work with new starter

### LINKED ISSUE #28986: UVE: Changes made in recent starter is breaking the examples
### Parent Issue

TBD 

### Problem Statement

We are having errors when setting up the UVE with recent starters.

Some errors that I found while trying to run UVE using demo site.

## Screenshots

Screenshot 2024-06-24 at 1 26 16 PM

You can see in the image above, that we are trying to render images from urls like: https://demo.dotcms.comcd6ddfb96c7dec2eb00c99c9d50a7e3f?language_id=1

There's a missing `/` and if we fix that in the code we can see that all the images are broken:

https://github.com/dotCMS/core/assets/63567962/5ee8996e-ce1b-4298-9941-1ad93f28e55b

In the case of https://demo.dotcms.comcd6ddfb96c7dec2eb00c99c9d50a7e3f?language_id=1

The correct way to see this image is by adding `/dA/` to the url like this: https://demo.dotcms.com/dA/cd6ddfb96c7dec2eb00c99c9d50a7e3f?language_id=1

We can also find this kind of errors with the most recent starter. We can get to broken links like: https://demo.dotcms.comundefinedf?language_id=1

This problems are cause by a change in the latest starter that involved the replacement of the `Binary Field` in all images for a `Image Field`. 

Both fields have a different way to handle the images and we developed our examples to work with the `Binary Field` that's why when we try to run the examples with an starter that has this changes. All images are broken or the example does not run at all.

Since the starter changed. We should adapt the components to work with the latest changes in the starter.

### Steps to Reproduce

Try to run any headless example using an environment that have the latest starter.

### Acceptance Criteria
Headless examples should be updated to work with the fields that changed in the most recent starter

### dotCMS Version

`latest-trunk`

### Proposed Objective

Technical User Experience

### Proposed Priority

Priority 1 - Show Stopper

### Assumptions & Initiation Needs

- You understand how to configure the UVE 
- You understand how to configure the Headless example 
- You understand how to change your custom starter 

### Quality Assurance Notes & Workarounds

No workarounds.

### PR DESCRIPTION
**Pull request by zJaaal**
### Proposed Changes

- Make the necessary changes to make headless work with starter from `[20240611]`

#### NextJS

- Add `imageLoader` function to handle image src build
- Update all `Image` components to use this new function
- Update all components to not access the context when not needed

https://github.com/dotCMS/core/assets/63567962/4a6c511c-f4b6-4cee-918d-9c747e4a4c25

#### Angular WIP

- Update current `imageLoader` to work with Image Field

https://github.com/dotCMS/core/assets/63567962/9f3ca714-2dc9-4c7e-b830-4440d96e1277

This PR fixes: #28986

### CODE REVIEW

File: examples/nextjs/src/components/content-types/activity.js [hunk 1]
```diff
@@ -14,9 +15,10 @@ function Activity({ title, description, image, urlTitle }) {
             {image && (
                 <Image
                     className="w-full"
-                    src={`${process.env.NEXT_PUBLIC_DOTCMS_HOST}${
-                        image?.idPath || image
-                    }?language_id=${language?.id}`}
+                    loader={imageLoaderByConfig({
+                        language: language?.id ?? 1,
+                    })}
```
Comment (fmontes) [inline examples/nextjs/src/components/content-types/activity.js:20]: Can't we make this global?

Comment (fmontes) [reply examples/nextjs/src/components/content-types/activity.js:20]: Don't want to have to pass in every `<Image />`

Comment (zJaaal) [reply examples/nextjs/src/components/content-types/activity.js:20]: Done, the only down side is that in the footer, we cannot get the current `languageId` but is something that we can solve later.

**Review by oidacra — APPROVED** [review]

**Review by nicobytes — APPROVED** [review]
LGTM, great support for custom img loader in NextJS

**Review by hmoreras — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

**Pull request merged**